jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java

```diff
@@ -539,12 +539,31 @@ protected void doStart() throws Exception
             if (getStopAtShutdown())
                 ShutdownThread.register(this);
 
-            //Register the Server with the handler thread for receiving
-            //remote stop commands
-            ShutdownMonitor.register(this);
+            // Look for the old-school (now deprecated) ShutdownMonitor
+            // First we look to see if anyone has called ShutdownMonitor.getInstance() yet.
+            //noinspection removal
+            ShutdownMonitor shutdownMonitor = ShutdownMonitor.INSTANCE.get();
+            //noinspection removal
+            if (shutdownMonitor != null && shutdownMonitor.getPort() != (-1))
+            {
+                addBean(shutdownMonitor);
+            }
 
-            //Start a thread waiting to receive "stop" commands.
-            ShutdownMonitor.getInstance().start(); // initialize
+            if (getBean(ShutdownService.class) == null)
```

Comment:
Isn't this going to result in both an old `ShutdownMonitor` and a `ShutdownService`?